rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy

```diff
@@ -3985,7 +3990,11 @@ class ScheduledExecutionService implements ApplicationContextAware, Initializing
                 def projectConfig = framework.frameworkProjectMgr.loadProjectConfig(scheduledExecution.project)
                 boolean disableRemoteOptionJsonCheck = projectConfig.hasProperty(REMOTE_OPTION_DISABLE_JSON_CHECK)
 
-                remoteResult = ScheduledExecutionController.getRemoteJSON(srcUrl, timeout, contimeout, retryCount, disableRemoteOptionJsonCheck)
+                remoteResult = ScheduledExecutionController.getRemoteJSON(new HttpClientCreator(){
+                    HttpClient<HttpResponse> createClient(){
+                        new ApacheHttpClient()
+                    }
```

Comment:
I think you could just pass a closure, `{->new ApacheHttpClient()}` or even just `ApacheHttpClient.&newInstance` instead of an anonymous class